### PR TITLE
feat(trait): let Camel runtime manages Kamelets

### LINKED ISSUE #4618: Let Camel framework manage Kamelets
Kamelets resolution is still done in the old fashioned way:
```
[1] 2023-07-27 07:24:31,641 INFO  [org.apa.cam.k.lis.SourcesConfigurer] (main) Loading routes from: SourceDefinition{name='camel-k-embedded-flow', language='yaml', type='source', location='file:/etc/camel/sources/camel-k-embedded-flow.yaml', }
[1] 2023-07-27 07:24:31,678 INFO  [org.apa.cam.k.lis.SourcesConfigurer] (main) Loading routes from: SourceDefinition{name='log-sink', language='yaml', type='source', location='file:/etc/camel/sources/log-sink.yaml', }
[1] 2023-07-27 07:24:31,683 INFO  [org.apa.cam.k.lis.SourcesConfigurer] (main) Loading routes from: SourceDefinition{name='timer-source', language='yaml', type='source', location='file:/etc/camel/sources/timer-source.yaml', }
```
The runtime takes care to do the work which supposedly has to be done by the framework. This was fine in the early days of Kamelets, but since now Kamelets are a Camel thing, we need to let it manage them accordingly.

### PR DESCRIPTION
**Pull request by squakez**
With this PR we get rid off the very first Kamelets runtime execution which translated the Kamelets into a route templates. We let now the Camel runtime to use the Kamelets as they are expected in the framework.

* Add a configmap as a bundle for all kamelets used by the Integration
* Mount the bundle in /etc/camel/kamelets
* Let the runtime use the Kamelets instead of trasforming into RouteTemplates
* Let the operator use the Kamelet template for capability parsing

Supersedes #4797 
Closes #4618 

**Release Note**
```release-note
feat(trait): let Camel runtime manages Kamelets
```

### CODE REVIEW

**Comment by squakez** [comment]
@christophd this PR address the point you raised about sources parsing. It should be solved now. Consider that the default properties are no longer required as they are managed by the runtime.

File: pkg/trait/camel.go [hunk 1]
```diff
@@ -238,8 +239,9 @@ func (t *camelTrait) computeConfigMaps(e *Environment) []ctrl.Object {
 		)
 	}
 
-	for i, s := range sources {
-		if s.ContentRef != "" || e.isEmbedded(s) {
+	i := 0
```
Comment (gansheer) [inline pkg/trait/camel.go:242]: A little nitpick out of curiosity: what is the root cause for this change of starting index value from 1 to 0 ?

**Review by gansheer — APPROVED** [review]

File: pkg/trait/camel.go (hunk 1 above)
Comment (squakez) [reply pkg/trait/camel.go:242]: the change is not the index itself. The problem is that if we let the variable as implicit part of the loop, it increments also when we skip the source, ie, is embedded, kamelet, etc. But we only want to increment when it is really processed, reason why we need to make it explicit.

Comment (gansheer) [reply pkg/trait/camel.go:242]: Thanks, forget the "1" part, I was mixing with the test result change.

**Comment by squakez** [comment]
Failure does not seem related.

**Pull request merged**